ui/components/DependenciesView.tsx

```diff
@@ -0,0 +1,141 @@
+      <Flex column align shadow>
+        <h2>No Dependencies</h2>
+
+        <div>
+          There are no dependencies set up for your kustomizations or helm
+          releases at this time. You can set them up using the dependsOn field
+          on the kustomization or helm release object.
+        </div>
+
+        <h2>What are dependencies for?</h2>
+
+        <div>
+          Dependencies allow you to relate different kustomizations and helm
+          releases, as well as specifying an order in which your resources
+          should be started. For example, you can wait for a database to report
+          as 'Ready' before attempting to deploy other services.
+        </div>
+      </Flex>
+    </Flex>
+  );
+}
+
+const Message = styled(UnstyledMessage)`
+  & {
+    ${Flex} {
+      box-sizing: border-box;
+      width: 560px;
+      padding-top: ${(props) => props.theme.spacing.medium};
+      padding-right: ${(props) => props.theme.spacing.xl};
+      padding-bottom: ${(props) => props.theme.spacing.xxl};
+      padding-left: ${(props) => props.theme.spacing.xl};
+      margin-top: ${(props) => props.theme.spacing.xxl};
+      margin-bottom: ${(props) => props.theme.spacing.xxl};
+      border-radius: 10px;
+      background-color: #ffffffd9;
+      color: ${(props) => props.theme.colors.neutral30};
+
+      h2 {
+        margin-top: 0;
+        margin-bottom: 0;
+        font-size: ${(props) => props.theme.fontSizes.large};
+        font-weight: 600;
+
+        &:not(:first-child) {
+          margin-top: ${(props) => props.theme.spacing.medium};
+        }
+      }
+
+      div {
+        margin-top: ${(props) => props.theme.spacing.medium};
+        font-size: 16px;
+        font-weight: 400;
+        line-height: 20px;
+      }
+    }
+  }
+`;
+
+type DependenciesViewProps = {
+  className?: string;
+  automation?: Automation;
+};
+
+const graphNodesPlaceholder = [] as FluxObjectNode[];
+
+function DependenciesView({ className, automation }: DependenciesViewProps) {
+  const [graphNodes, setGraphNodes] = React.useState<FluxObjectNode[] | null>(
+    null
+  );
+
+  const automationKind = automation?.kind;
+
+  const {
+    data,
+    isLoading: isLoadingData,
+    error,
+  } = automation
+    ? useListObjects("", fluxObjectKindToKind(automationKind))
+    : { data: { objects: [], errors: [] }, error: null, isLoading: false };
+
+  React.useEffect(() => {
+    if (isLoadingData) {
+      return;
+    }
+
+    if (error || data?.errors?.length > 0) {
+      setGraphNodes(graphNodesPlaceholder);
+      return;
+    }
+
+    const nodes = getGraphNodes(
+      data.objects.map((obj) => new FluxObjectNode(obj)),
+      automation
+    );
+
+    if (nodes.length === 0) {
+      setGraphNodes(graphNodesPlaceholder);
+    } else {
+      setGraphNodes(nodes);
+    }
+  }, [isLoadingData, data, error]);
+
+  const isLoading = isLoadingData && !graphNodes;
+
+  const shouldShowGraph = !!graphNodes && graphNodes.length > 0;
```

Comment:
This component will render the Message component if the data finished loading (= graphNodes have been set to a non-null value no matter if there is an error or not, but if there was an error they will be an empty array) and the length of graphNodes is 0. Before the data finished loading the component should not display the No Dependencies message.

Reply:
Ah, you probably meant not to show the No Dependencies message if there is an error?

Reply:
So, if the error happens, it can still mean that there are dependencies, just smth. happened while retrieving the objects. We don't know if the object is in a dependencies tree until we load all objects of this kind and build the tree, because the object can be a root node and itself have no dependencies.

So, for now I'll keep displaying the No Dependencies message if there is an error when loading data, probably. TO provide some information to the user. @ozamosi should I display the No Dependencies message if an error happened or should I display nothing? The actual error message will be displayed by the wrapper `RequestStateHandler` component.

Reply:
This is OK as it is.

Reply:
OK, good.

Reply:
Ah, no, checked how RequestStateHandler component works. It will not show any wrapped content if there is an error. It returns earlier with the error message Alert. It's a pity.